src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -159,15 +149,14 @@ public void Enqueue(TElement element, TPriority priority)
             _size++;
             _version++;
 
-            // Restore the heap order
-            int lastNodeIndex = GetLastNodeIndex();
+            (TElement Element, TPriority Priority) item = (element, priority);
             if (_comparer == null)
             {
-                MoveUpDefaultComparer((element, priority), lastNodeIndex);
+                MoveUpDefaultComparer(ref item, _size - 1);
```

Comment:
I tried making similar ref changes and didn't see it have a measurable impact on string,string or int,int.  Do you see a benefit?  I imagine the benefit may be larger for larger data types; did you try that?

Reply:
There are marginal improvements, but largely the impact in the benchmarks came from moving to a binary heap. I will need to include more benchmarks using struct types.

Reply:
> largely the impact in the benchmarks came from moving to a binary heap

That would be my expectation as well.

I suggest separating the changes and measuring them on their own merit separately.